libraries/exoplayer/src/main/java/androidx/media3/exoplayer/audio/AudioCapabilities.java

```diff
@@ -220,7 +234,13 @@ public Pair<Integer, Integer> getEncodingAndChannelConfigForPassthrough(Format f
       channelCount = getMaxSupportedChannelCountForPassthrough(encoding, sampleRate);
     } else {
       channelCount = format.channelCount;
-      if (channelCount > maxChannelCount) {
+      // Some DTS:X TVs reports ACTION_HDMI_AUDIO_PLUG.EXTRA_MAX_CHANNEL_COUNT as 8
+      // instead of 10. See https://github.com/androidx/media/issues/396
+      if (format.sampleMimeType.equals(MimeTypes.AUDIO_DTS_X)) {
```

Comment:
@tianyif I do not have access to other device at the moment but it is reasonable to think that this would be reproduceable for all DTS:X TVs based on MTK platforms. It will take sometime for this issue to be fixed on the platform. Meanwhile, Xperi will have to go to each and every streaming service to tell them to add the above so that DTS:X can be played back. I think it is better to add this patch in the exoplayer code base so that everyone will be aware of it.  When the issue get fixed on the platform (which could take a long time), we can remove this patch in another exoplayer contribution. Thanks.

Reply:
I will keep this patch. Thanks!